### PR TITLE
Overhaul FE JS and implement timestamps

### PR DESCRIPTION
**Pull request by rijkvanzanten**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,138 +1,178 @@
...
 
-  // Register service worker
-  if ('serviceWorker' in navigator) {
-    // Where to find the service worker
-    navigator.serviceWorker.register('/sw.js');
-  }
+/**
+ * Hides the typing indicator
+ */
+function hideLoader() {
```
Comment (dandevri) [inline src/index.js:169]: Not sure if these two needs to be in index.js? `hideloader` and `showloader`.  
Even `./utils` isn't the right fit. ¯\_(ツ)_/¯

File: server/message-handler.js [hunk 2]
```diff
@@ -11,13 +11,17 @@ module.exports = function(io) {
...
       'Hallo, ik ben <strong>Robat</strong>. Door <strong>help</strong> te typen kan je zien wat je mij kan vragen!',
-      'Hoi, ik ben Robat! Weten wat je mij kan vragen? Typ dan <strong>help</strong.',
+      'Hoi, ik ben Robat! Weten wat je mij kan vragen? Typ dan <strong>help</strong>.',
     ];
 
     // Generate a random number based on the length of the array
-    const introMessage = introMessages[Math.floor(Math.random()*introMessages.length)];
+    const introMessage = {
+      value: introMessages[Math.floor(Math.random() * introMessages.length)],
+      id: -1,
```
Comment (dandevri) [inline server/message-handler.js:20]: ID was `0` and now it is `-1`. Just personal preference?

File: src/utils.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ * Returns date in YYYY-MM-DD
+ * @param  {Date} date Date to convert
+ * @return {String} Date in YYYY-MM-DD
+ */
+export function formatDate(date) {
+  if (typeof date === 'string') {
+    date = new Date(date);
+  }
+
+  return `${date.getFullYear()}-${('0' + (date.getMonth() + 1)).slice(-2)}-${('0' + date.getDate()).slice(-2)}`;
```
Comment (dandevri) [inline src/utils.js:11]: Wow such amazing. 🎉   Smart use `.slice(-2)`

File: server/message-handler.js (hunk 2 above)
Comment (rijkvanzanten) [reply server/message-handler.js:20]: -1 > 0

**Review by dandevri — APPROVED** [review]
Most of it makes sense. Better structured 🎉
Some things I don't understand but let's look at this next week. Because I want to go home and have a pizza and some beer. I am done, I don't even want to review you're pull request I just want to go home. Everybody wants to go home. Mirza wants to go home, giulia wants to go home and Pierre is already at home.

**Comment by rijkvanzanten** [comment]
fair enough :+1:

**Pull request merged**